plugins/client/graphql-kotlin-client-generator/src/main/kotlin/com/expediagroup/graphql/plugin/client/generator/types/generateTypeName.kt

```diff
@@ -258,3 +300,40 @@ private fun calculateSelectedFields(
     }
     return result
 }
+
+/**
+ * Determines if a GraphQL object type should be created as a shared response type.
+ * This simply checks if the feature is enabled - all ObjectTypeDefinition types are candidates for reuse.
+ */
+private fun shouldCreateSharedResponseType(context: GraphQLClientGeneratorContext, typeName: String): Boolean {
```

Comment:
the existence of this function looks a little off, why the typeName is passed around argument if not used, maybe just do an inline boolean check